pkg/registry/generic/etcd/etcd.go

```diff
@@ -279,19 +279,19 @@ func (e *Etcd) Update(ctx api.Context, obj runtime.Object) (runtime.Object, bool
 			if err != nil {
 				return nil, nil, err
 			}
-		} else {
-			// Check if the object's resource version matches the latest resource version.
-			newVersion, err := e.Storage.Versioner().ObjectResourceVersion(obj)
-			if err != nil {
-				return nil, nil, err
-			}
-			if newVersion != version {
-				return nil, nil, kubeerr.NewConflict(e.QualifiedResource, name, fmt.Errorf("the object has been modified; please apply your changes to the latest version and try again"))
-			}
 		}
 		if err := rest.BeforeUpdate(e.UpdateStrategy, ctx, obj, existing); err != nil {
 			return nil, nil, err
 		}
+		// Check if the object's resource version matches the latest resource version.
```

Comment:
This change is actually wrong. If another party has updated the object, the server should return NewConflict, but now probably a validation error like "only xxx field can be updated" will be returned. I'll revise the code.